### ISSUE TITLE
[Bug]: Ruby version not supported

### ISSUE THREAD

**Issue by adamchipperfield** [Shopify/cli#3502]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

App, Extension

### Expected behavior

Running `yarn dev` on an app should start development mode

### Actual behavior

Running `yarn dev` briefly starts development mode, but fails with an error after a few seconds:

>Ruby version 2.6.10 is not supported
>Make sure you have at least Ruby 2.7.5 installed on your system. [Documentation](https://www.ruby-lang.org/en/documentation/installation/).

I uninstalled Shopify CLI and re-installed using brew, since Shopify states it will install the required Ruby version, but still no luck. I also cannot upgrade the system Ruby.

### Verbose output

```shell
Not really sure what to put here. I ran it with `--verbose` but it just output tons of info (100s and 100s of lines that seemed unrelated).
```

### Reproduction steps

1. In an app, run `yarn dev`

### Operating System

Mac OS Ventura 13.6.1

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.56.3

### Node version (run `node -v` if you're not sure)

v18.19.0

**Comment by shauns** [comment]
There's a limit to what our tooling can offer, and managing your systems ruby installation unfortunately sits outside of it. I haven't seen the specific issue you encountered before. You might have better luck managing your system ruby via something like `chruby`. The CLI will use whichever ruby is first on the path.

**Comment by bansavage** [comment]
@adamchipperfield Did you discover a fix? having similar issue

**Comment by shauns** [comment]
I appreciate the annoyance, fighting with package managers is nobodies idea of a good time and I have very much been there before. There's a limit on what we can offer as it is environment specific: there are plenty of devs using the ruby-dependent features of the CLI right now.

I recommend chruby as its what we use internally at Shopify (which is very much a Ruby shop) to manage multiple Ruby versions when that's needed. It can be installed via homebrew itself, and then chruby can take care of installing and switching different ruby versions, leaving your system one intact.

More generally we are obviously (follow the commits!) increasingly leaning on JS based tooling for things like the CLI, to try and help avoid this sort of scenario. But Ruby is still required today.

**Comment by bansavage** [comment]
@shauns 
Do you recommended installing Ruby with chruby first? THEN installing the CLI? Or would this result in 2 incompatible Ruby versions?

I'm also facing the Bundler gem error using theme dev commands.
The Shopify docs state: `Note: When you install Shopify CLI using Homebrew, Homebrew installs Node.js, Ruby, and Git for you.`

**Comment by adamchipperfield** [comment]
I appreciate this is closed, but to address the suggestion to **just use chruby**:

I installed chruby and managed to change my Ruby version. I start the app and avoid the Ruby version errors. Great!

But now I get issues around Bundler not being installed. I follow the suggestion and ran `gem install bundler`. I get more errors that I don't understand, and am now having to waste client hours researching what Bundler even is.

So it's all good and well stepping outside of the Liquid comfort zone _when things go smoothly_. But any slightly advanced error and I'm stuck.

I appreciate the team are moving to a JS based tooling, but I take issue with offering this as a way to build JS and Liquid extensions and not disclosing the buckets of Ruby knowledge you'll need.

**Comment by adamchipperfield** [comment]
Cheers for the update! I've updated to the latest version and now I'm getting this issue when working with themes, so it's completely debilitating.

> Then reinstall the CLI.

Could you show exactly how we'd do this ^ please?

**Comment by bansavage** [comment]
```
brew uninstall shopify-cli
brew install shopify-cli
```

**Comment by MauroLZ** [comment]
> Cheers for the update! I've updated to the latest version and now I'm getting this issue when working with themes, so it's completely debilitating.
> 
> > Then reinstall the CLI.
> 
> Could you show exactly how we'd do this ^ please?

Hey Adam! I ran into the same issue today after trying to develop a simple app, while my usual work is as a theme dev.
What I believe was happening is that the Ruby version on my M1 mac was overriding the one I installed using rbenv, so even after setting the local and global version to 3.3.0, using 'ruby -v' still showed 2.6.1

What fixed it for me was to change to the Homebrew installed version by changing the path:
echo 'export PATH="/usr/local/opt/ruby/bin:$PATH"' >> ~/.zshrc 

Then applying the changes using:
source ~/.zshrc

Now running 'ruby -v' on my project terminal shows 3.3.0. Hopefully that does it for you, its very frustrating, I know.
